opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -531,7 +531,9 @@ def start_span(  # pylint: disable=too-many-locals
         if parent_context is not None and not isinstance(
             parent_context, trace_api.SpanContext
         ):
-            raise TypeError
+            raise TypeError(
+                "expected Span, SpanContext or None object for trace parent."
```

Comment:
What's the difference between trace parent and span parent? I thought we were restricting the span parents to be only spancontexts?

Reply:
I don't think there's technically a "trace parent": A trace is a collection of correlated spans. I'll fix the error. 

The parent can still *accept* both a span or a spancontext, but either way the span will have a spancontext assigned to it. 

The error message is designed to accurately message back to the user of the *_span methods that parent can be a Span, a SpanContext, or None.

Reply:
I think "trace parent" is confusing because it makes me think of the W3C traceparent header which is kinda related but not exactly what is meant here. Thus I suggest

```suggestion
                "parent must be a Span, SpanContext or None."
```

Just use exactly the same name as the parameter name in the function definition.